src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/IUnconfiguredProjectCommonServices.cs

```diff
@@ -1,88 +1,32 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.ComponentModel.Composition;
 using Microsoft.VisualStudio.ProjectSystem;
 using Microsoft.VisualStudio.ProjectSystem.References;
 
 namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
 {
-    [Export(typeof(IUnconfiguredProjectCommonServices))]
-    internal class UnconfiguredProjectCommonServices : IUnconfiguredProjectCommonServices
+    // This defines the set of services that we frequently need for working with UnconfiguredProject.
+    //
+    // We're following a somewhat common pattern for code that uses CPS. It's really easy to end up
+    // relying on service location inside CPS, which can be hard to test. This approach makes it easy
+    // for us to build reusable mocks instead.
+    internal interface IUnconfiguredProjectCommonServices
```

Comment:
Ignore the diff on this file. I had to swap the names of the interface and the implementation